frontend/src/app/components/groups/move-group-members/move-group-members.component.ts

```diff
@@ -0,0 +1,204 @@
+    this.userService.getMultipleByIds(memberIDs).then((users) => {
+      if (users) this.unassigned.push(...users);
+      this.updateGroups.emit(this.getGroups());
+    });
+  }
+
+  private getGroups(): Group[] {
+    const groups: Group[] = [];
+    this.groupMembers.forEach((gm) => {
+      const updatedGroup: Group = {
+        groupID: gm.groupID,
+        projectID: this.projectID,
+        name: gm.groupName,
+        members: gm.members.map((member) => member.userID),
+      };
+      groups.push(updatedGroup);
+    });
+    return groups;
+  }
+
+  unassignAllMembers() {
+    this.clearAllGroupMembers();
+    this.updateUnassignedMembers();
+  }
+
+  removeMember(groupIndex: number, memberIndex: number): void {
+    let members = this.groupMembers[groupIndex].members;
+    this.unassigned.push(members[memberIndex]);
+    members.splice(memberIndex, 1);
+    this.updateGroups.emit(this.getGroups());
+  }
+
+  removeAllMembers(groupIndex: number): void {
+    let members = this.groupMembers[groupIndex].members;
+    if (members.length === 0) return;
+    this.unassigned.push(...members);
+    members.length = 0;
+    this.updateGroups.emit(this.getGroups());
+  }
+
+  shuffleBetweenGroups(): void {
+    if (this.groupMembers.length == 0) return;
+
+    const members = this.removeDuplicateUsers(
+      this.getAllMembersInSelectedGroups()
+    );
+    const shuffledMembers = this.shuffleArray(members);
+
+    this.clearUnassignedMembers();
+    this.clearAllGroupMembers();
+
+    shuffledMembers.forEach((member, index) => {
+      this.groupMembers[index % this.groupMembers.length].members.push(member);
+    });
+    this.updateGroups.emit(this.getGroups());
+  }
+
+  private clearUnassignedMembers(): void {
+    this.unassigned = [];
+  }
+
+  private clearAllGroupMembers(): void {
+    this.groupMembers.forEach((group) => (group.members = []));
+  }
+
+  private getAllMembersInSelectedGroups(): User[] {
+    const members: User[] = [];
+    this.groupMembers.forEach((group) => {
+      members.push(...group.members);
+    });
+    members.push(...this.unassigned);
+    return members;
+  }
+
+  private removeDuplicateUsers(users: User[]): User[] {
+    const ids = users.map((user) => user.userID);
+    return users.filter(
+      ({ userID }, index) => !ids.includes(userID, index + 1)
+    );
+  }
+
+  drop(event: CdkDragDrop<User[]>) {
+    if (event.previousContainer === event.container) {
+      moveItemInArray(
+        event.container.data,
+        event.previousIndex,
+        event.currentIndex
+      );
+    } else {
+      transferArrayItem(
+        event.previousContainer.data,
+        event.container.data,
+        event.previousIndex,
+        event.currentIndex
+      );
+      this.updateGroups.emit(this.getGroups());
+    }
+  }
+
+  // https://stackoverflow.com/a/2450976/4970939
+  // Uses Fisher-Yates shuffle https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
+  private shuffleArray(array: any[]): any[] {
```

Comment:
We already have a utility function for shuffling arrays inside `utils/Utils.ts`. Either you can re-use that one and remove this, or you can add this shuffle function to the `utils/Utils.ts` if it's important we use this specific algorithm and then call it from here (`Utils.fisherShuffle(..)`).

Reply:
Nice, I'll use the function from utils. The algorithm used was just the one used in SCORE.